controllers/datadogagent/feature/types.go

```diff
@@ -16,27 +17,69 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// ComponentsEnabled use to know which component need to be enabled for the feature
+// If set to:
+//   * true: the feature needs the corresponding component.
+//   * false: the corresponding component needs to ne disabled for this feature.
+//   * nil: the feature doesn't need the corresponding component.
+type ComponentsEnabled struct {
+	ClusterAgent       *bool
+	Agent              *bool
+	ClusterCheckRunner *bool
+}
+
+// IsEnabled return true if the Feature need to be enabled
+func (cc *ComponentsEnabled) IsEnabled() bool {
+	return apiutils.BoolValue(cc.ClusterAgent) || apiutils.BoolValue(cc.Agent) || apiutils.BoolValue(cc.ClusterCheckRunner)
```

Comment:
the cluster checks runner would never be enabled without the cluster agent, right? maybe the last `||` is not needed

Reply:
it was to have something generic.
my idea was: if a feature require the clustercheckrunner it should deploy the clusteragent too.